src/pks/biogeochemistry/bgc_simple/bgc_simple.cc

```diff
@@ -517,6 +517,16 @@ void BGCSimple::FieldToColumn_(AmanziMesh::Entity_ID col, const Epetra_Vector& v
   }
 }
 
+// helper function for pushing field to column
+void BGCSimple::FieldToColumn_(AmanziMesh::Entity_ID col, const Epetra_Vector& vec,
+                               double* col_vec, int ncol) {
+  ColIterator col_iter(*mesh_, mesh_surf_->entity_get_parent(AmanziMesh::CELL, col), ncol);
+  for (std::size_t i=0; i!=col_iter.size(); ++i) {
+    col_vec[i] = vec[col_iter[i]];
```

Comment:
this could just be:

    auto& col_iter = mesh_->cells_of_column(col);
    for (...) {
      col_vec[i] = vec[col_iter[i]]
    }

And remove the ColIterator object which is not needed.